timeseries/src/autogluon/timeseries/models/local/statsmodels.py

```diff
@@ -54,74 +82,71 @@ class ETSModel(AbstractStatsmodelsModel):
         When set to -1, all CPU cores are used.
     """
 
-    quantile_method_name = "pred_int"
-    statsmodels_allowed_init_args = [
+    allowed_local_model_args = [
         "error",
         "trend",
         "damped_trend",
         "seasonal",
         "seasonal_period",
-    ]
-    statsmodels_allowed_fit_args = [
         "maxiter",
     ]
 
-    def _update_sm_model_init_args(self, sm_model_init_args: dict, data: TimeSeriesDataFrame) -> dict:
-        sm_model_init_args = sm_model_init_args.copy()
-        sm_model_init_args["freq"] = data.freq
-        sm_model_init_args.setdefault("trend", "add")
-
-        # Infer seasonal_period if seasonal_period is not given / is set to None
-        seasonal_period = sm_model_init_args.pop("seasonal_period", None)
-        if seasonal_period is None:
-            seasonal_period = get_seasonality(data.freq)
-        sm_model_init_args["seasonal_periods"] = seasonal_period
+    def _update_local_model_args(
+        self, local_model_args: Dict[str, Any], data: TimeSeriesDataFrame, **kwargs
+    ) -> Dict[str, Any]:
+        local_model_args.setdefault("trend", "add")
+        local_model_args.setdefault("maxiter", 1000)
 
-        seasonal = sm_model_init_args.setdefault("seasonal", "add")
-        # Disable seasonality if seasonal_period is too short
+        seasonal_period = local_model_args.pop("seasonal_period")
+        seasonal = local_model_args.setdefault("seasonal", "add")
         if seasonal is not None and seasonal_period <= 1:
             logger.warning(
                 f"{self.name} with seasonal = {seasonal} requires seasonal_period > 1 "
                 f"(received seasonal_period = {seasonal_period}). Disabling seasonality."
             )
-            sm_model_init_args["seasonal"] = None
-            sm_model_init_args["seasonal_periods"] = 1
+            local_model_args["seasonal"] = None
+            local_model_args["seasonal_periods"] = 1
```

Comment:
`seasonal_period` and `seasonal_periods` look a bit dangerous. Should we at least mark it with a TODO to rename here?

Reply:
Every method in `statsmodels` has a different name for this. We chose to always expose `seasonal_period` to the user in our API, and internally map this to the model's parameter (`seasonal_periods`, `period` or an entry in `seasonal_order` tuple). Not sure if we can change this. The name `seasonal_period` is also used in our 0.5.2 release and in Darts, so I'm not included to change this.